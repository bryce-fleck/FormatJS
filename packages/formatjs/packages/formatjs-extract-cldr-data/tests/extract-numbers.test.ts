import extractNumbers from '../src/extract-numbers';

describe('extract-numbers', function() {
  it('should be able to extract en', function() {
    expect(extractNumbers(['en'])).toMatchSnapshot();
  });
});
