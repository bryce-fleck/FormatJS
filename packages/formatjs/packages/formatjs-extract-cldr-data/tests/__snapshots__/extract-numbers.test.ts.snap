// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract-numbers should be able to extract en 1`] = `
Object {
  "en": Object {
    "currencyFormats-numberSystem-latn": Object {
      "accounting": "¤#,##0.00;(¤#,##0.00)",
      "currencySpacing": Object {
        "afterCurrency": Object {
          "currencyMatch": "[:^S:]",
          "insertBetween": " ",
          "surroundingMatch": "[:digit:]",
        },
        "beforeCurrency": Object {
          "currencyMatch": "[:^S:]",
          "insertBetween": " ",
          "surroundingMatch": "[:digit:]",
        },
      },
      "short": Object {
        "standard": Object {
          "1000-count-one": "¤0K",
          "1000-count-other": "¤0K",
          "10000-count-one": "¤00K",
          "10000-count-other": "¤00K",
          "100000-count-one": "¤000K",
          "100000-count-other": "¤000K",
          "1000000-count-one": "¤0M",
          "1000000-count-other": "¤0M",
          "10000000-count-one": "¤00M",
          "10000000-count-other": "¤00M",
          "100000000-count-one": "¤000M",
          "100000000-count-other": "¤000M",
          "1000000000-count-one": "¤0B",
          "1000000000-count-other": "¤0B",
          "10000000000-count-one": "¤00B",
          "10000000000-count-other": "¤00B",
          "100000000000-count-one": "¤000B",
          "100000000000-count-other": "¤000B",
          "1000000000000-count-one": "¤0T",
          "1000000000000-count-other": "¤0T",
          "10000000000000-count-one": "¤00T",
          "10000000000000-count-other": "¤00T",
          "100000000000000-count-one": "¤000T",
          "100000000000000-count-other": "¤000T",
        },
      },
      "standard": "¤#,##0.00",
      "unitPattern-count-one": "{0} {1}",
      "unitPattern-count-other": "{0} {1}",
    },
    "decimalFormats-numberSystem-latn": Object {
      "long": Object {
        "decimalFormat": Object {
          "1000-count-one": "0 thousand",
          "1000-count-other": "0 thousand",
          "10000-count-one": "00 thousand",
          "10000-count-other": "00 thousand",
          "100000-count-one": "000 thousand",
          "100000-count-other": "000 thousand",
          "1000000-count-one": "0 million",
          "1000000-count-other": "0 million",
          "10000000-count-one": "00 million",
          "10000000-count-other": "00 million",
          "100000000-count-one": "000 million",
          "100000000-count-other": "000 million",
          "1000000000-count-one": "0 billion",
          "1000000000-count-other": "0 billion",
          "10000000000-count-one": "00 billion",
          "10000000000-count-other": "00 billion",
          "100000000000-count-one": "000 billion",
          "100000000000-count-other": "000 billion",
          "1000000000000-count-one": "0 trillion",
          "1000000000000-count-other": "0 trillion",
          "10000000000000-count-one": "00 trillion",
          "10000000000000-count-other": "00 trillion",
          "100000000000000-count-one": "000 trillion",
          "100000000000000-count-other": "000 trillion",
        },
      },
      "short": Object {
        "decimalFormat": Object {
          "1000-count-one": "0K",
          "1000-count-other": "0K",
          "10000-count-one": "00K",
          "10000-count-other": "00K",
          "100000-count-one": "000K",
          "100000-count-other": "000K",
          "1000000-count-one": "0M",
          "1000000-count-other": "0M",
          "10000000-count-one": "00M",
          "10000000-count-other": "00M",
          "100000000-count-one": "000M",
          "100000000-count-other": "000M",
          "1000000000-count-one": "0B",
          "1000000000-count-other": "0B",
          "10000000000-count-one": "00B",
          "10000000000-count-other": "00B",
          "100000000000-count-one": "000B",
          "100000000000-count-other": "000B",
          "1000000000000-count-one": "0T",
          "1000000000000-count-other": "0T",
          "10000000000000-count-one": "00T",
          "10000000000000-count-other": "00T",
          "100000000000000-count-one": "000T",
          "100000000000000-count-other": "000T",
        },
      },
      "standard": "#,##0.###",
    },
    "defaultNumberingSystem": "latn",
    "minimalPairs": Object {
      "few": "Take the {0}rd right.",
      "one": "Take the {0}st right.",
      "other": "Take the {0}th right.",
      "pluralMinimalPairs-count-one": "{0} day",
      "pluralMinimalPairs-count-other": "{0} days",
      "two": "Take the {0}nd right.",
    },
    "minimumGroupingDigits": "1",
    "miscPatterns-numberSystem-latn": Object {
      "approximately": "~{0}",
      "atLeast": "{0}+",
      "atMost": "≤{0}",
      "range": "{0}–{1}",
    },
    "otherNumberingSystems": Object {
      "native": "latn",
    },
    "percentFormats-numberSystem-latn": Object {
      "standard": "#,##0%",
    },
    "scientificFormats-numberSystem-latn": Object {
      "standard": "#E0",
    },
    "symbols-numberSystem-latn": Object {
      "decimal": ".",
      "exponential": "E",
      "group": ",",
      "infinity": "∞",
      "list": ";",
      "minusSign": "-",
      "nan": "NaN",
      "perMille": "‰",
      "percentSign": "%",
      "plusSign": "+",
      "superscriptingExponent": "×",
      "timeSeparator": ":",
    },
  },
}
`;

exports[`extract-numbers should be able to extract vi 1`] = `
Object {
  "vi": Object {
    "currencyFormats-numberSystem-latn": Object {
      "accounting": "#,##0.00 ¤",
      "currencySpacing": Object {
        "afterCurrency": Object {
          "currencyMatch": "[:^S:]",
          "insertBetween": " ",
          "surroundingMatch": "[:digit:]",
        },
        "beforeCurrency": Object {
          "currencyMatch": "[:^S:]",
          "insertBetween": " ",
          "surroundingMatch": "[:digit:]",
        },
      },
      "short": Object {
        "standard": Object {
          "1000-count-other": "0 N ¤",
          "10000-count-other": "00 N ¤",
          "100000-count-other": "000 N ¤",
          "1000000-count-other": "0 Tr ¤",
          "10000000-count-other": "00 Tr ¤",
          "100000000-count-other": "000 Tr ¤",
          "1000000000-count-other": "0 T ¤",
          "10000000000-count-other": "00 T ¤",
          "100000000000-count-other": "000 T ¤",
          "1000000000000-count-other": "0 NT ¤",
          "10000000000000-count-other": "00 NT ¤",
          "100000000000000-count-other": "000 NT ¤",
        },
      },
      "standard": "#,##0.00 ¤",
      "unitPattern-count-other": "{0} {1}",
    },
    "decimalFormats-numberSystem-latn": Object {
      "long": Object {
        "decimalFormat": Object {
          "1000-count-other": "0 nghìn",
          "10000-count-other": "00 nghìn",
          "100000-count-other": "000 nghìn",
          "1000000-count-other": "0 triệu",
          "10000000-count-other": "00 triệu",
          "100000000-count-other": "000 triệu",
          "1000000000-count-other": "0 tỷ",
          "10000000000-count-other": "00 tỷ",
          "100000000000-count-other": "000 tỷ",
          "1000000000000-count-other": "0 nghìn tỷ",
          "10000000000000-count-other": "00 nghìn tỷ",
          "100000000000000-count-other": "000 nghìn tỷ",
        },
      },
      "short": Object {
        "decimalFormat": Object {
          "1000-count-other": "0 N",
          "10000-count-other": "00 N",
          "100000-count-other": "000 N",
          "1000000-count-other": "0 Tr",
          "10000000-count-other": "00 Tr",
          "100000000-count-other": "000 Tr",
          "1000000000-count-other": "0 T",
          "10000000000-count-other": "00 T",
          "100000000000-count-other": "000 T",
          "1000000000000-count-other": "0 NT",
          "10000000000000-count-other": "00 NT",
          "100000000000000-count-other": "000 NT",
        },
      },
      "standard": "#,##0.###",
    },
    "defaultNumberingSystem": "latn",
    "minimalPairs": Object {
      "one": "Rẽ vào lối rẽ thứ nhất bên phải.",
      "other": "Rẽ vào lối rẽ thứ {0} bên phải.",
      "pluralMinimalPairs-count-one": "Rẽ vào lối rẽ thứ nhất bên phải.",
      "pluralMinimalPairs-count-other": "Rẽ vào lối rẽ thứ {0} bên phải.",
    },
    "minimumGroupingDigits": "1",
    "miscPatterns-numberSystem-latn": Object {
      "approximately": "~{0}",
      "atLeast": "{0}+",
      "atMost": "≤{0}",
      "range": "{0}-{1}",
    },
    "otherNumberingSystems": Object {
      "native": "latn",
    },
    "percentFormats-numberSystem-latn": Object {
      "standard": "#,##0%",
    },
    "scientificFormats-numberSystem-latn": Object {
      "standard": "#E0",
    },
    "symbols-numberSystem-latn": Object {
      "decimal": ",",
      "exponential": "E",
      "group": ".",
      "infinity": "∞",
      "list": ";",
      "minusSign": "-",
      "nan": "NaN",
      "perMille": "‰",
      "percentSign": "%",
      "plusSign": "+",
      "superscriptingExponent": "×",
      "timeSeparator": ":",
    },
  },
}
`;
